File: codeforces/64.required_remainder/required_remainder.py
# Author: José Rodolfo (jric2002)
t = int(input())
while (t):
  x, y, n = list(map(int, input().split(" ")))
  c = int(n / x)
  while (True):
    k = (x * c) + y
    if (k <= n):
      break
    c -= 1
  print(k)
  t -= 1

File: codeforces/low_difficulty/800_points/64.required_remainder_[1374A]/required_remainder.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned int t, x, y, n, c, k;
  cin >> t;
  while (t--) {
    cin >> x >> y >> n;
    c = (n / x);
    while (true) {
      k = ( x * c) + y;
      if (k <= n) {
        break;
      }
      c -= 1;
    }
    cout << k << endl;
  }
  return 0;
}